src/pybind/mgr/dashboard/frontend/src/app/core/error/error.component.ts

```diff
@@ -0,0 +1,45 @@
+import { Location } from '@angular/common';
+import { Component, OnInit } from '@angular/core';
+import { ActivatedRoute } from '@angular/router';
+
+@Component({
+  selector: 'cd-error',
+  templateUrl: './error.component.html',
+  styleUrls: ['./error.component.scss']
+})
+export class ErrorComponent implements OnInit {
+  topHeader: string;
+  topMessage: string;
+  btmHeader: string;
+  btmMessage: string;
+  section: string;
+  section_info: string;
+  fa_icon: string;
+
+  constructor(public route: ActivatedRoute, private location: Location) {}
+
+  ngOnInit() {
+    try {
+      this.topMessage = history.state.topMessage;
+      this.topHeader = history.state.topHeader;
+      this.btmHeader = history.state.btmHeader;
+      this.btmMessage = history.state.btmMessage;
+      this.section = history.state.section;
+      this.section_info = history.state.section_info;
+      this.fa_icon = history.state.fa_icon;
+
+      if (this.topMessage === '' || this.topMessage === undefined) {
+        this.topHeader = $localize`Client Error`;
+        this.topMessage = $localize`Sorry for the inconvinience, we have encountered a Client Error.`;
+        this.btmHeader = $localize`Page Not Found`;
+        this.btmMessage = $localize`Sorry, we couldn’t find what you were looking for.
+            The page you requested may have been changed or moved.`;
+        this.fa_icon = 'fa fa-exclamation-circle';
+      }
+    } catch (error) {}
+  }
+
+  back() {
+    this.location.back();
+  }
```

Comment:
No need of wrapping this. You can put methods in HTML templates as long as they are in event handlers.